File: src/main/java/com/gioia/radiogio/services/PlayerServiceImpl.java
package com.gioia.radiogio.services;

import com.gioia.radiogio.data.domains.RadioStation;
import com.gioia.radiogio.data.repositories.OldRadioStationRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.config.ConfigurableBeanFactory;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Service;
import uk.co.caprica.vlcj.player.component.AudioPlayerComponent;

@Service
@Scope(ConfigurableBeanFactory.SCOPE_SINGLETON)
public class PlayerServiceImpl implements PlayerService {
    //@Autowired
    private AudioPlayerComponent audioPlayerComponent;

    @Autowired
    private OldRadioStationRepository oldRadioStationRepository;

    private Logger logger = LoggerFactory.getLogger(PlayerServiceImpl.class);

    @Override
    public void playRadio(String url){
        audioPlayerComponent.mediaPlayer().media().play(url);
    }

    @Override
    public void stopRadio(){
        audioPlayerComponent.mediaPlayer().controls().pause();
    }

    @Override
    public void toggleFavourite(Boolean isNowInFavourite, RadioStation radioStation) {
        if(radioStation == null) return;

        if(isNowInFavourite){
            logger.atDebug().log("Añandiendo a favoritos la radio" + radioStation.getName());
        }
        else {
            logger.atDebug().log("Eliminando de favoritos la radio" + radioStation.getName());
        }

        radioStation.setIsFavourite(!radioStation.getIsFavourite());
        oldRadioStationRepository.updateOne(radioStation);
    }

    @Override
    public void changeVolume(Integer value){
        audioPlayerComponent.mediaPlayer().audio().setVolume(value);
    }
}

File: src/main/kotlin/com/gioia/radio/services/PlayerServiceImpl.kt
package com.gioia.radio.services

import com.gioia.radio.data.domains.RadioStation
import com.gioia.radio.data.repositories.RadioStationRepository
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import uk.co.caprica.vlcj.player.component.AudioPlayerComponent

class PlayerServiceImpl(
    private var audioPlayerComponent: AudioPlayerComponent,
    private var radioStationRepository: RadioStationRepository
) : PlayerService {
    private var logger: Logger = LoggerFactory.getLogger(PlayerServiceImpl::class.java)

    override fun playRadio(url: String){
        audioPlayerComponent.mediaPlayer().media().play(url)
    }

    override fun stopRadio(){
        audioPlayerComponent.mediaPlayer().controls().pause()
    }

    override fun toggleFavourite(isNowInFavourite: Boolean, radioStation: RadioStation?) {
        if(radioStation == null) return
        logger.atDebug().log("${if(isNowInFavourite) "Añandiendo a" else "Eliminando de"} favoritos la radio ${radioStation.name}")
        radioStation.isFavourite = !radioStation.isFavourite
        radioStationRepository.updateOne(radioStation)
    }

    override fun changeVolume(value: Int){
        audioPlayerComponent.mediaPlayer().audio().setVolume(value)
    }
}